pyvista/core/dataobject.py

```diff
@@ -161,8 +161,8 @@ def _write_vtk(mesh_: DataObject) -> None:
             set_vtkwriter_mode(vtk_writer=writer, use_binary=binary)
             writer.SetFileName(str(file_path))
             writer.SetInputData(mesh_)
-            if isinstance(writer, _vtk.vtkPLYWriter) and texture is not None:
-                mesh_ = cast(pyvista.DataSet, mesh_)
+            if isinstance(writer, _vtk.vtkPLYWriter) and texture is not None:  # type: ignore[unreachable]
+                mesh_ = cast(pyvista.DataSet, mesh_)  # type: ignore[unreachable]
```

Comment:
This is OK. These annotations were added recently and I recall the typing here was challenging (hence the cast, too)